### PR TITLE
Add license-check plugin bootstrap

### PR DESCRIPTION
**Pull request by kungfoo**
This is adding bootstrap stuff for having the license check/report plugin that @godrja wants to work on.

Might want to do this on this branch and then merge.

### CODE REVIEW

File: beekeeper-license-check/src/main/groovy/io/beekeeper/gradle/licenses/LicenseCheckPlugin.groovy [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.github.jk1.license.LicenseReportExtension
+import com.github.jk1.license.LicenseReportPlugin
+import com.github.jk1.license.ProjectData
+import com.github.jk1.license.reader.ProjectReader
+
+import groovy.transform.CompileStatic
+
+@CompileStatic
+class LicenseCheckPlugin implements Plugin<Project> {
```
Comment (Wattos) [inline beekeeper-license-check/src/main/groovy/io/beekeeper/gradle/licenses/LicenseCheckPlugin.groovy]: as discussed, any new plugin should be written in java

File: beekeeper-formatter-plugin/src/main/groovy/io/beekeeper/formatter/FormatterPlugin.groovy [hunk 2]
```diff
@@ -34,10 +34,32 @@ class FormatterPlugin implements Plugin<Project> {
 
 	void applySpotlessConfiguration(Project project) {
 		project.spotless {
-			java {
-				removeUnusedImports()
-				trimTrailingWhitespace()
-				eclipse().configFile getJavaFormattingRulesAbsoluteProjectPath(project)
+			groovyGradle {
```
Comment (Wattos) [inline beekeeper-formatter-plugin/src/main/groovy/io/beekeeper/formatter/FormatterPlugin.groovy]: These changes should not be part of this pull request

**Comment by Wattos** [comment]
@kungfoo Please remove any changes which are not part of the license check. Please make sure we add new plugins in java as discussed

File: beekeeper-license-check/src/main/groovy/io/beekeeper/gradle/licenses/LicenseCheckPlugin.groovy (hunk 1 above)
Comment (kungfoo) [reply beekeeper-license-check/src/main/groovy/io/beekeeper/gradle/licenses/LicenseCheckPlugin.groovy]: Then I'll convert the file.

File: beekeeper-formatter-plugin/src/main/groovy/io/beekeeper/formatter/FormatterPlugin.groovy (hunk 2 above)
Comment (kungfoo) [reply beekeeper-formatter-plugin/src/main/groovy/io/beekeeper/formatter/FormatterPlugin.groovy]: Yes I will clean up this mess.

File: beekeeper-formatter-plugin/build.gradle [hunk 3]
```diff
@@ -1,3 +1,6 @@
+apply plugin: 'java-gradle-plugin'
+apply plugin: 'com.gradle.plugin-publish'
```
Comment (Wattos) [inline beekeeper-formatter-plugin/build.gradle]: happens automatically from the parent plugin

File: beekeeper-license-check/build.gradle [hunk 4]
```diff
@@ -0,0 +1,15 @@
+apply plugin: 'java-gradle-plugin'
+apply plugin: 'com.gradle.plugin-publish'
```
Comment (Wattos) [inline beekeeper-license-check/build.gradle]: from parent

File: build.gradle [hunk 5]
```diff
@@ -28,10 +26,15 @@ subprojects {
...
+        
+        maven {
+            url 'https://plugins.gradle.org/m2/'
+        }
     }
 
     dependencies {
-        implementation localGroovy()
         testImplementation "org.spockframework:spock-core:1.2-groovy-2.5"
+        implementation localGroovy()
```
Comment (Wattos) [inline build.gradle]: implementation always above testImplementation

File: settings.gradle [hunk 6]
```diff
@@ -2,3 +2,5 @@ rootProject.name = 'beekeeper-gradle-plugins'
 
 include ':beekeeper-formatter-plugin'
 include ':beekeeper-plugin'
+include ':beekeeper-license-check'
```
Comment (Wattos) [inline settings.gradle]: inconsistent naming

File: build.gradle [hunk 7]
```diff
@@ -16,8 +16,6 @@ plugins {
 subprojects {
 
     apply plugin: 'groovy'
-    apply plugin: 'java-gradle-plugin'
```
Comment (Wattos) [inline build.gradle]: we could apply convention over configuration.

if (it.name.endsWith "-plugin") {
   apply plugin: 'java-gradle-plugin'
   apply plugin: 'com.gradle.plugin-publish'
}

File: settings.gradle (hunk 6 above)
Comment (Wattos) [reply settings.gradle]: `beekeeper-license-check-plugin`

File: settings.gradle [hunk 8]
```diff
@@ -2,3 +2,5 @@ rootProject.name = 'beekeeper-gradle-plugins'
...
+include ':plugin-test-helpers'
```
Comment (Wattos) [inline settings.gradle]: order should be:

```
include ':beekeeper-plugin-test-helpers'

include ':beekeeper-formatter-plugin'
include ':beekeeper-license-check'
include ':beekeeper-plugin'
```

File: plugin-test-helpers/build.gradle [hunk 9]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (Wattos) [inline plugin-test-helpers/build.gradle]: rename to

```
beekeeper-test-support
```

Comment (kungfoo) [reply plugin-test-helpers/build.gradle]: `beekeeper-plugin-test-support`?

File: build.gradle (hunk 7 above)
Comment (kungfoo) [reply build.gradle]: Done.

**Comment by kungfoo** [comment]
Should address most if not all of the review points.

**Comment by kungfoo** [comment]
We should also come up with a eclipse formatter profile we can use for groovy files. The default `greclipse()` formatter uses tabs instead of spaces. Which of course makes a mess of everything.

File: beekeeper-formatter-plugin/src/main/groovy/io/beekeeper/formatter/FormatterPlugin.groovy [hunk 10]
```diff
@@ -33,11 +33,14 @@ class FormatterPlugin implements Plugin<Project> {
 	}
 
 	void applySpotlessConfiguration(Project project) {
-		project.spotless {
-			java {
-				removeUnusedImports()
-				trimTrailingWhitespace()
-				eclipse().configFile getJavaFormattingRulesAbsoluteProjectPath(project)
+		if(project.plugins.findPlugin("java") != null) {
```
Comment (Wattos) [inline beekeeper-formatter-plugin/src/main/groovy/io/beekeeper/formatter/FormatterPlugin.groovy]: not needed anymore, fixed in new java version

Comment (Wattos) [reply beekeeper-formatter-plugin/src/main/groovy/io/beekeeper/formatter/FormatterPlugin.groovy]: Also, turned out that this was not the correct approach. We need to use withPlugin() as the java plugin might not yet be applied

File: beekeeper-license-check-plugin/build.gradle [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+	implementation group: 'com.github.jk1', name: 'gradle-license-report', version: '1.6'
+}
+
+gradlePlugin {
+    plugins {
+        licenseCheckPlugin {
+            id = 'io.beekeeper.gradle.plugins.license-check'
+            implementationClass = 'io.beekeeper.gradle.licenses.LicenseCheckPlugin'
+        }
+    }
```
Comment (Wattos) [inline beekeeper-license-check-plugin/build.gradle]: Must apply mavenCoordinates, otherwise the plugin will not be found

```
mavenCoordinates {
        groupId = project.group
    }
```

File: settings.gradle [hunk 12]
```diff
@@ -1,4 +1,7 @@
 rootProject.name = 'beekeeper-gradle-plugins'
 
+include ':beekeeper-plugin-test-support'
+
 include ':beekeeper-formatter-plugin'
 include ':beekeeper-plugin'
+include ':beekeeper-license-check-plugin'
```
Comment (Wattos) [inline settings.gradle]: move above `beekeeper-plugin`

File: beekeeper-license-check-plugin/build.gradle (hunk 11 above)
Comment (kungfoo) [reply beekeeper-license-check-plugin/build.gradle]: Will do.

File: settings.gradle (hunk 12 above)
Comment (kungfoo) [reply settings.gradle]: Will do.

File: beekeeper-formatter-plugin/src/main/groovy/io/beekeeper/formatter/FormatterPlugin.groovy (hunk 10 above)
Comment (kungfoo) [reply beekeeper-formatter-plugin/src/main/groovy/io/beekeeper/formatter/FormatterPlugin.groovy]: True. Will change this code.
